### PR TITLE
Update model to use true carbon values when available.

### PR DESCRIPTION
**Pull request by VimWizardVersace**
The model uses ground truth carbon values when available, and fills in the gaps by querying the carbon isoscape for missing values. 

Right now, the columns to combine queried and groundtruth are hardcoded. There is a more dignified way to do this, certainly, but i wanted to minimize the diff here.

Some other notable diffs:
Main uses kernel initializers. This version of the model did not, and it's not currently clear what the effect of glorot initializers has on it.
Main has a patience of 100. This version of the model has a patience of 1000.

### CODE REVIEW

File: dnn/variational_inference.ipynb
Comment (rothn) [inline dnn/variational_inference.ipynb:1]: Does our currently-checked-in pipeline have "d13C_cel_mean_artificial"?

How is this different from "d13C_cel_mean"?

Also, it seems like both versions of the code load "d13C_cel_map_BRAZIL_stack.tiff" but this version calls it "d13C_cel_mean_artificial" and the current main calls it "d13C_cel_mean". Did you intend to make additional changes?

Comment (rothn) [inline dnn/variational_inference.ipynb:1]: Why do we set `n=50` here for `sample_normal_distribution()`? Didn't we discuss n=5? Did this change at some point?

Comment (rothn) [inline dnn/variational_inference.ipynb:1]: I see `patience=1000` here. Why such a high patience setting? This seems very likely to overfit.

**Review by rothn — CHANGES_REQUESTED** [review]
I've left some more detail-oriented comments, but overall, I don't see any high-level semantic changes here.

The results recorded here do not match what the current pipeline achieves as far as I can tell. Some of the issues might be the data.

Is this the right version of the notebook? Or are you also using some custom data to get these results that is different from what the data pipeline in main does?

File: dnn/variational_inference.ipynb
Comment (VimWizardVersace) [reply dnn/variational_inference.ipynb:1]: At the time I created the model, `main` had patience=1000

Comment (VimWizardVersace) [reply dnn/variational_inference.ipynb:1]: d13C_cel_mean_artificial -> d13C_cel values queried from a raster. It's not a real column in the training data.
d13C_cel -> ground truth carbon
d13C_cel_mean -> both of the above combined, prioritizing d13C_cel

Comment (VimWizardVersace) [reply dnn/variational_inference.ipynb:1]: Main had n=15 at the time. I changed it to 50 for this model in particular to smoothen the loss curves. Model quality varied a lot between seeds if I didn't do this.

Comment (rothn) [reply dnn/variational_inference.ipynb:1]: So, seems like you're saying that using the raster produced better results than using the ground-truth carbon values. Am I understanding correctly?

Is it intentional that the variable name is different but the raster file these come from is the same in both notebooks?

Comment (rothn) [reply dnn/variational_inference.ipynb:1]: What I think you're saying is that here we perform additional data preprocessing to train with cellulose means from the ground truth when we have them, which did not exist in the old version of the notebook. Resolving, but please re-open if this is incorrect.
